tensornetwork/contractors/custom_path_solvers/pathsolvers.py

```diff
@@ -0,0 +1,414 @@
+# Copyright 2019 The TensorNetwork Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+import numpy as np
+from typing import List, Union, Any, Tuple, Type, Optional, Sequence
+
+
+#------------------------------------------------------------------------#
+def greedy_size_solve(log_adj_in: np.ndarray):
+  """
+  Solve for the contraction order of a tensor network (encoded as a 
+  log-adjacency matrix) using a greedy algorithm that minimizes the 
+  intermediate tensor sizes. 
+  Args:
+    log_adj_in: matrix where element [i,j] is the log10 of the total dimension 
+      of the indices connecting ith and jth tensors.
+  Returns:
+    np.ndarray: cheapest contraction order found, specified as a sequence of 
+      binary contractions.
+    float: the cost of the network contraction, given as log10(total_FLOPS).
+  """  
+  N = log_adj_in.shape[0]
+  log_adj = log_adj_in.copy().reshape(N,N)
+  orders = np.zeros([2,0],dtype=int)
+  costs = None;
+  
+  for t in range(N-1):
```

Comment:
The range of the loop here is `N`, which is the same name that you are overwriting below. It's not wrong, but I would still try to avoid it (confusing)